src/kaitai_spec/attribute/tests.rs

```diff
@@ -238,30 +239,29 @@ fn simple_enum2() {
         doc_ref: None,
         type_: AttributeType::Enumeration(enumeration::Enumeration {
             name: String::from("ip_prot"),
-            type_: enumeration::EnumType::S1,
+            type_: common::Integer::S1,
         }),
     };
     assert_eq!(_deserialized, expect);
 }
 
 #[test]
-#[should_panic]
 fn incorrect_type_enum1() {
     let str: &str = r#"
         id: birth_year
         enum: MA1
         type: incorrect
         doc: My doc"#;
-    let _deserialized: Attribute = serde_yaml::from_str(str).expect("Failed!");
+    let _deserialized: Result<Attribute, _> = serde_yaml::from_str(str);
```

Comment:
add assert!(is_err()) and here it is not necessary to start variable name with an underscore